types/src/main.ts

```diff
@@ -1350,6 +1436,14 @@ export interface CodeActionContext {
 	 * An array of diagnostics.
 	 */
 	diagnostics: Diagnostic[];
+
+	/**
+	 * Requested kind of actions to return.
+	 *
+	 * Actions not of this kind are filtered out by the client before being shown. So servers
+	 * can omit computing them.
+	 */
+	only?: CodeActionKind
```

Comment:
Why isn't this a `CodeActionKind[]` so that the client can request two separate kinds of code actions? What if I don't want `quickfix` but I want both `refactor` and `source`?